### PR TITLE
Remove fake "ajax" module

### PR DESCRIPTION
**Pull request by lucaswoj**
This PR removes the fake `js/util/ajax` module, instead leveraging Sinon's fake [`XMLHttpRequest`](http://sinonjs.org/docs/#server) functionality. Because of this change

 - `js/util/browser/ajax.js` is fully tested (and moved to `js/util/ajax.js`)
 - there are no 404 error messages during unit testing
 - there is no need to start / stop real HTTP servers during testing
 - we have the ability to create cheap mock HTTP servers using Sinon

cc @jfirebaugh @mollymerp 

ref https://github.com/mapbox/mapbox-gl-js/issues/3121

## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] diagnose test-suite hangs 
 - [x] ensure unit test error messages are suppressed 
 - [ ] get a PR review

### CODE REVIEW

**Comment by lucaswoj** [comment]
This is ready for 👀 @jfirebaugh

File: test/suite_implementation.js [hunk 1]
```diff
@@ -12,6 +17,7 @@ module.exports = function(style, options, _callback) {
         }
     }
 
+    window.XMLHttpRequest = XMLHttpRequest;
```
Comment (jfirebaugh) [inline test/suite_implementation.js]: Is there a reason the default jsdom XHR isn't suitable? Can we document it with a comment?

File: test/js/source/vector_tile_source.test.js [hunk 2]
```diff
@@ -1,13 +1,9 @@
 'use strict';
 
 var test = require('tap').test;
-var st = require('st');
```
Comment (jfirebaugh) [inline test/js/source/vector_tile_source.test.js]: Should be able to remove this from devDependencies as well.

File: test/suite_implementation.js (hunk 1 above)
Comment (jfirebaugh) [inline test/suite_implementation.js]: Oh, I see, it's overwriting the sinon fake XHR. Can we use jsdom XHR by default and only substitute sinon for the unit tests?

Comment (lucaswoj) [inline test/suite_implementation.js]: The built-in jsdom XHR doesn't work but I don't fully understand why. I'll dig in now.

**Comment by lucaswoj** [comment]
I listed my 4 reasons "why" in the original PR. Stubing 2 of the 4 methods when we run `test-suite` is an improvement over never testing the `ajax` module 😄.

**Comment by mourner** [comment]
@lucaswoj no, I was referring to `st` module removal. It's used in npm scripts for running a debug/bench server.

**Comment by lucaswoj** [comment]
Oops! You're right. I didn't think of that. Restoring shortly.

**Comment by mourner** [comment]
@lucaswoj do you know why the test suite runs ~4 minutes slower in this branch, or is this some kind of a random Circle deviation?

**Comment by lucaswoj** [comment]
@mourner I haven't looked into it yet. There were a few layers of abstraction added into the networking stack. Perhaps we should mock out the entire AJAX module as before during the test suite run.

**Comment by jfirebaugh** [comment]
Looking at recent successful builds, the length of this one doesn't seem to be too far outside the normal variance of test times.

Good to merge from my perspective, and a much-welcome cleanup.

**Comment by mourner** [comment]
@jfirebaugh it looks quite far outside to me — consistently 16-17m for test.sh (after rerunning it a few times) where all other recent builds are 11-13m.

**Comment by lucaswoj** [comment]
The tests themselves aren't taking longer. However there is some background work preventing the process from exiting once the tests finish.

**Comment by lucaswoj** [comment]
JSDom's XHR library is causing the process to hang. 

The most effective use of our time is to stub out all methods on the `ajax` module when running the render tests. We may want to spend time debugging JSDom and submitting an upstream patch in the future.

**Comment by lucaswoj** [comment]
Ready to 🚢?

**Comment by jfirebaugh** [comment]
🚢 

I have some pending ajax.js changes that will as a side-effect make stubbing it out cleaner.

**Pull request merged**